include/seqan3/alignment/scoring/detail/simd_matrix_scoring_scheme.hpp

```diff
@@ -0,0 +1,190 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::simd_matrix_scoring_scheme.
+ * \author Joshua Kim <joshua.kim AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/alignment/configuration/align_config_mode.hpp>
+#include <seqan3/alignment/scoring/scoring_scheme_concept.hpp>
+#include <seqan3/alphabet/concept.hpp>
+#include <seqan3/core/concept/cereal.hpp>
+#include <seqan3/core/simd/concept.hpp>
+#include <seqan3/core/simd/simd_algorithm.hpp>
+#include <seqan3/std/concepts>
+
+#include <seqan3/core/debug_stream.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief A vectorised scoring scheme to handle scoring matrices.
+ * \ingroup scoring
+ * \tparam simd_score_t The type of the simd vector; must model seqan3::detail::simd_concept.
+ * \tparam alphabet_t The type of the alphabet over which to define the scoring scheme; must model seqan3::semialphabet
+ *                    and must have an alphabet size of at least 2.
```

Comment:
Not that it would be a relevant use case but in this general case you can also have single character alphabet. So I would remove the restriction here.